libcrux-ml-dsa/src/simd/avx2/arithmetic.rs

```diff
@@ -59,3 +59,18 @@ pub fn montgomery_multiply(lhs: Vec256, rhs: Vec256) -> Vec256 {
 
     mm256_sub_epi32(product_high, c)
 }
+
+#[inline(always)]
+pub fn shift_left_then_reduce(simd_unit: Vec256, shift_by: usize) -> Vec256 {
+    // TODO: Shift using slli
```

Comment:
Not quite, the problem is in order to use slli we'd need to pass the shift amount as a const-generic parameter, which would then mean adding a const generic parameter to `shift_left_then_reduce`, which would cause some upstream changes, but I can make them now.

Reply:
Done.